contracts/locked/LockedSOV.sol

```diff
@@ -0,0 +1,371 @@
+	 */
+	constructor(
+		address _SOV,
+		address _vestingRegistry,
+		uint256 _cliff,
+		uint256 _duration,
+		address[] memory _admins
+	) public {
+		require(_SOV != address(0), "Invalid SOV Address.");
+		require(_vestingRegistry != address(0), "Vesting registry address is invalid.");
+
+		SOV = IERC20(_SOV);
+		vestingRegistry = VestingRegistry(_vestingRegistry);
+		cliff = _cliff * fourWeeks();
+		duration = _duration * fourWeeks();
+
+		for (uint256 index = 0; index < _admins.length; index++) {
+			isAdmin[_admins[index]] = true;
+		}
+	}
+
+	/* Public or External Functions */
+
+	/**
+	 * @notice The function to add a new admin.
+	 * @param _newAdmin The address of the new admin.
+	 * @dev Only callable by an Admin.
+	 */
+	function addAdmin(address _newAdmin) public onlyAdmin {
+		require(_newAdmin != address(0), "Invalid Address");
+		require(!isAdmin[_newAdmin], "Address is already admin");
+		isAdmin[_newAdmin] = true;
+
+		emit AdminAdded(msg.sender, _newAdmin);
+	}
+
+	/**
+	 * @notice The function to remove an admin.
+	 * @param _adminToRemove The address of the admin which should be removed.
+	 * @dev Only callable by an Admin.
+	 */
+	function removeAdmin(address _adminToRemove) public onlyAdmin {
+		require(isAdmin[_adminToRemove], "Address is not an admin");
+		isAdmin[_adminToRemove] = false;
+
+		emit AdminRemoved(msg.sender, _adminToRemove);
+	}
+
+	/**
+	 * @notice The function to update the Vesting Registry, Duration and Cliff.
+	 * @param _vestingRegistry The Vesting Registry Address.
+	 * @param _cliff The time period after which the tokens begin to unlock.
+	 * @param _duration The time period after all tokens will have been unlocked.
+	 * @dev IMPORTANT 1: You have to change Vesting Registry if you want to change Duration and/or Cliff.
+	 * IMPORTANT 2: `_cliff` and `_duration` is multiplied by 4 weeks in this function.
+	 */
+	function changeRegistryCliffAndDuration(
+		address _vestingRegistry,
+		uint256 _cliff,
+		uint256 _duration
+	) external onlyAdmin {
+		require(address(vestingRegistry) != _vestingRegistry, "Vesting Registry has to be different for changing duration and cliff.");
+		/// If duration is also zero, then it is similar to Unlocked SOV.
+		require(_duration != 0, "Duration cannot be zero.");
+
+		vestingRegistry = VestingRegistry(_vestingRegistry);
+
+		cliff = _cliff * fourWeeks();
+		duration = _duration * fourWeeks();
+
+		emit RegistryCliffAndDurationUpdated(msg.sender, _vestingRegistry, _cliff, _duration);
+	}
+
+	/**
+	 * @notice Adds SOV to the user balance (Locked and Unlocked Balance based on `_basisPoint`).
+	 * @param _userAddress The user whose locked balance has to be updated with `_sovAmount`.
+	 * @param _sovAmount The amount of SOV to be added to the locked and/or unlocked balance.
+	 * @param _basisPoint The % (in Basis Point)which determines how much will be unlocked immediately.
+	 */
+	function deposit(
+		address _userAddress,
+		uint256 _sovAmount,
+		uint256 _basisPoint
```

Comment:
any user can call `LockedSOV.deposit(addr, 1, <inf>)` to immediately unlock all `addr`'s tokens, effectively disabling `lockedBalances` logic
PS: `<inf> == LockedSOV.getLockedBalance(addr) + 1`

Reply:
Nice catch. Added in e38c7bb